src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/EETypeParameterSymbol.cs

```diff
@@ -59,6 +59,14 @@ public override bool HasReferenceTypeConstraint
             get { return _sourceTypeParameter.HasReferenceTypeConstraint; }
         }
 
+        public override bool IsReferenceTypeFromConstraintTypes
+        {
+            get
+            {
+                return _sourceTypeParameter.IsReferenceTypeFromConstraintTypes || CalculateIsReferenceTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics);
```

Comment:
>|| CalculateIsReferenceTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics) [](start = 79, length = 83)

Is this necessary?

Reply:
> Is this necessary?

Yes, due to possible type substitution involved new types can appear in the constraint types

---
In reply to: [521608683](https://github.com/dotnet/roslyn/pull/49261#discussion_r521608683) [](ancestors = 521608683)